dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -660,6 +670,47 @@ private void loadMetadata(UfsStatus status, List<LoadFailure> errors) {
     }
   }
 
+  private void loadPages(String ufsPath, List<PageId> pageIds, long fileLength)
+      throws AccessControlException, IOException {
+    Optional<UnderFileSystem> ufs = mUfsManager.get(new AlluxioURI(ufsPath));
+    if (!ufs.isPresent()) {
+      throw new RuntimeException("Ufs not found for " + ufsPath);
+    }
+    long lastPageId = fileLength / mPageSize;
+    // TODO(elega) can batch multiple pages together ot speed up the loading.
+    for (PageId pageId : pageIds) {
+      int lengthToLoad = (int)
+          (pageId.getPageIndex() == lastPageId ? fileLength % mPageSize : mPageSize);
+      long offset = pageId.getPageIndex() * mPageSize;
+      ByteBuf buf = PooledByteBufAllocator.DEFAULT.directBuffer(lengthToLoad);
```

Comment:
Can we avoid allocating buffer inside the for loop?
Move this allocation of memory out of the loop. One page size of memory as the default and max value.